Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py

```diff
@@ -1821,6 +1864,222 @@ def test_module(client: Client, *_) -> Tuple[str, Dict[Any, Any], Dict[Any, Any]
+
+        comments_context = {'comments_and_work_notes': note.get('value')}
+        entries.append({
+            'Type': note.get('type'),
+            'Category': note.get('category'),
+            'Contents': note.get('value'),
+            'ContentsFormat': note.get('format'),
+            'Tags': note.get('tags'),
+            'Note': True,
+            'EntryContext': comments_context
+        })
+    # Parse user dict to email
+    assigned_to = ticket.get('assigned_to', {})
+    caller = ticket.get('caller_id', {})
+    if assigned_to:
+        user_result = client.get('sys_user', assigned_to.get('value'))
+        user = user_result.get('result', {})
+        user_email = user.get('email')
+        ticket['assigned_to'] = user_email
+
+    if caller:
+        user_result = client.get('sys_user', caller.get('value'))
+        user = user_result.get('result', {})
+        user_email = user.get('email')
+        ticket['caller_id'] = user_email
+
+    if ticket.get('resolved_by'):
+        if params.get('close_incident'):
+            demisto.debug(f'ticket is closed: {ticket}')
+            entries.append({
+                'Type': EntryType.NOTE,
+                'Contents': {
+                    'dbotIncidentClose': True,
+                    'closeReason': f'From ServiceNow: {ticket.get("close_notes")}'
+                },
+                'ContentsFormat': EntryFormat.JSON
+            })
+
+    demisto.debug(f'Pull result is {ticket}')
+    return [ticket] + entries
+
+
+def update_remote_system_command(client: Client, args: Dict[str, Any], params: Dict[str, Any]) -> str:
+    """
+    This command pushes local changes to the remote system.
+    Args:
+        client:  XSOAR Client to use.
+        args:
+            args['data']: the data to send to the remote system
+            args['entries']: the entries to send to the remote system
+            args['incident_changed']: boolean telling us if the local incident indeed changed or not
+            args['remote_incident_id']: the remote incident id
+        params:
+            entry_tags: the tags to pass to the entries (to separate between comments and work_notes)
+
+    Returns: The remote incident id - ticket_id
+
+    """
+    parsed_args = UpdateRemoteSystemArgs(args)
+    if parsed_args.delta:
+        demisto.debug(f'Got the following delta keys {str(list(parsed_args.delta.keys()))}')
+
+    entry_tags = params.get('entry_tags', 'work_notes,comments').split(",")
+    ticket_type = client.ticket_type
+    ticket_id = parsed_args.remote_incident_id
+    if parsed_args.incident_changed:
+        demisto.debug(f'Incident changed: {parsed_args.incident_changed}')
+        fields = get_ticket_fields(parsed_args.data, ticket_type=ticket_type)
+        if not params.get('close_ticket'):
+            update_fields = {key: val for key, val in fields.items() if key != 'closed_at' and key != 'resolved_at'}
+            result = client.update(ticket_type, ticket_id, update_fields, {})
```

Comment:
use the condition only to set the `update_fields`, no need to call `update` in both
also need to change log level to `debug` from `info` and print it in both cases of the condition and not only in the `else` branch